### ISSUE TITLE
Put panel controls into panel headings

### ISSUE THREAD

**Issue by gordonwoodhull** [att/rcloud#463]
To conserve vertical space, and to emphasize their interactivity.

For search and help (#432).

**Comment by gordonwoodhull** [comment]
@shaneporter, the heart of this is to apply the same extension mechanism to open up the headers / headings of the panels for customization. Some automatic layout like what we're doing in the notebook cell headers, except here there's an extra dimension which is that all panels have different controls, maybe similar to the way the menus are extensions of extensions (or maybe not similar).

**Comment by shaneporter** [comment]
Taking 'Help' as an example, the header would contain the 'get help on...' input/button _and_ the Keyboard link to show the shortcuts? But then when you click on the '?' button, it would open the panel (if closed) and then show the help details?

For a simpler example, such as the 'clear session pane' (#1734), wouldn't the inclusion of a button make it difficult to distinguish between the icon and the button?

There's also, as you say, the issue of dealing with this additional content when the width of the column falls below a certain size. If it isn't shown, then it's gone, meaning that it can't be used at narrow column widths.

**Comment by gordonwoodhull** [comment]
Yes, that's the idea. Our buttons are mostly bright blue, so I won't worry about them being confused with the white-on-grey icons.

That's a good point about column widths. It will have to influence the "natural width" (`colwidth`) of the panel - so, if the panel is currently narrower than its `colwidth`, it's okay if the controls don't show.

But this does mean we'll have to implement some kind of hiding/collapsing. It was probably too ambitious for me to put this on 1.6.1 - guess I was just enthusiastic about the session clear button. We'll just have to figure out somewhere else to put that.
